renderdoc/driver/shaders/spirv/spirv_disassemble.cpp

```diff
@@ -297,7 +297,8 @@ rdcstr Reflector::Disassemble(const rdcstr &entryPoint,
       "Generator: %s\n"
       "Generator Version: %u\n"
       "\n",
-      m_MajorVersion, m_MinorVersion, m_SPIRV[3], ToStr(m_Generator).c_str(), m_GeneratorVersion);
+      m_MajorVersion, m_MinorVersion, m_SPIRV.size() > 3 ? m_SPIRV[3] : 0,
```

Comment:
What's the purpose behind this change, to prevent crashes if invalid SPIR-V is passed in? This should really be checked for elsewhere and earlier as I wouldn't want to have checks for `m_SPIRV` throughout this code.

Reply:
It is not checked, that is why there segfault with latest release when you try to debug shader object.

```c++
shadRefl.PopulateDisassembly(shader->spirv);
```
is invoked unconditionally (almost, `shadRefl.refl->debugInfo.debuggable` is true for some reason, probably it is also bug), it contains following code:
```c++
if(disassembly.empty())
    disassembly = spirv.Disassemble(refl->entryPoint, instructionLines);
```
disassembly is empty, so it reaches to spirv.Disassemble and there m_SPIRV dereference without any checks

Reply:
I understand but is that only as a result of the earlier problem that you've fixed where an invalid/unbound shader module is referenced when using the shader object extension? with your fix, does this crash still happen and if so from where?

As I say this should be fixed elsewhere if it's a problem - we should never get as far as trying to disassemble with an invalid shader module so it suggests the root problem is earlier. The crash you're describing is the symptom, not the problem

Reply:
> with your fix, does this crash still happen and if so from where?

No, there without surprises, it is dead code with fix

> As I say this should be fixed elsewhere if it's a problem - we should never get as far as trying to disassemble with an invalid shader module so it suggests the root problem is earlier. The crash you're describing is the symptom, not the problem

OK, if you think that 0 bounds is not good enough to make error visible, let me revert this change